src/ai/backend/agent/agent.py

```diff
@@ -696,6 +702,51 @@ async def produce_error_event(
         pretty_tb = "".join(traceback.format_tb(tb)).strip()
         await self.produce_event(AgentErrorEvent(pretty_message, pretty_tb))
 
+    async def _report_all_kernel_commit_status_map(self) -> None:
+        """
+        Commit statuses are managed by `lock` file.
+        +- base_commit_path
+        |_ subdir1 (usually user's email)
+            |_ commit_file1 (named by timestamp)
+            |_ commit_file2
+            |_ lock
+                |_ kernel_id1 (means the user is currently committing the kernel)
+                |_ kernel_id2
+        |_ subdir2
+        ...
+
+        `status_map` is like below.
+        {
+            str(subdir1): {
+                str(kernel_id1): CommitStatus.ONGOING.value,
+                str(kernel_id2): CommitStatus.ONGOING.value,
+            },
+            str(subdir2): ...,
+        }
+        """
+        base_commit_path: Path = self.local_config["agent"]["image-commit-path"]
+        status_map = {}
+        for subdir in base_commit_path.iterdir():
+            ongoing_list = [commit_path.name for commit_path in subdir.glob("./**/lock/*")]
```

Comment:
I know it is really easy to forget, but please note that `Path.glob` also is a blocking function. Depending on total count of committed containers and disk performance this can cause a potential performance issue, so let's wrap `subdir.glob` with `loop.run_in_executor()`.

Reply:
update this part. thank you!